tools/cdi-containerimage-server/main.go

```diff
@@ -46,8 +49,21 @@ func main() {
 	if err != nil {
 		log.Fatalf("Failed get image filename in %s: %v", *directory, err)
 	}
+
+	mux := http.NewServeMux()
+	mux.HandleFunc("/info", func(w http.ResponseWriter, _ *http.Request) {
+		info := common.ServerInfo{
+			Env: os.Environ(),
+		}
+		w.Header().Set("Content-Type", "application/json")
+		err := json.NewEncoder(w).Encode(info)
+		if err != nil {
+			http.Error(w, err.Error(), http.StatusInternalServerError)
```

Comment:
Not sure you can return the status code after you started writing to the response

Reply:
It looks like a commonly used pattern to me. Also, I took a peek into the code of `Encode` and it uses an internal buffer and only writes to an `io.Writer` when marshalling succeeded and no more errors can occur.

See https://cs.opensource.google/go/go/+/refs/tags/go1.22.1:src/encoding/json/stream.go;drc=dac9b9ddbd5160c5f4552410f5f8281bd5eed38c;l=230

Reply:
Not sure what happens in case of failures when writing to the response though? But I'm not sure if in that case there is anything left to do with the response.

Reply:
More hair splitting but if Encode is buffering everything then it would be better to call Marshall ourselves making it more likely to send the correct response code

Reply:
It is hair splitting, to be honest I'd like to keep it as is.